### PR TITLE
feat(legacy): Check for required fields in PET sidecars.

### PR DESCRIPTION
**Pull request by nellh**
Fix for #1953 by testing for the required fields when we examine the PET image sidecars.

```
        1: [ERR] A PET image JSON sidecar is missing a required field. (code: 237 - PET_JSON_KEY_REQUIRED)
                ./sub-01/ses-postitbs/pet/sub-01_ses-postitbs_pet.nii
                        You must define TracerName for this file. It can be included one of the following locations: /pet.json, /sub-01/sub-01_pet.json, /sub-01/ses-postitbs/sub-01_ses-postitbs_pet.json, /sub-01/ses-postitbs/pet/sub-01_ses-postitbs_pet.json
```

### CODE REVIEW

File: bids-validator/validators/nifti/nii.js [hunk 1]
```diff
@@ -1266,9 +1266,77 @@ export default function NIFTI(
...
+    'SpecificRadioactivity',
+    'SpecificRadioactivityUnits',
+    'ModeOfAdministration',
+    'TimeZero',
+    'ScanStart',
+    'InjectionStart',
+    'FrameTimesStart',
+    'FrameDuration',
+    'AcquisitionMode',
+    'ImageDecayCorrected',
```
Comment (effigies) [inline bids-validator/validators/nifti/nii.js:1297]:
```suggestion
    'ImageDecayCorrected',
    'ImageDecayCorrectionTime',
```

File: bids-validator/validators/nifti/nii.js [hunk 2]
```diff
@@ -1266,9 +1266,77 @@ export default function NIFTI(
...
+      'InfusionRadioactivity',
+      'InfusionStart',
+      'InfusionSpeed',
+      'InfusionSpeedUnits',
+      'InjectedVolume',
+    )
+  }
+  if (
+    mergedDictionary.hasOwnProperty('ReconFilterType') &&
+    mergedDictionary['ReconFilterType'] !== 'none'
```
Comment (effigies) [inline bids-validator/validators/nifti/nii.js]: Array of strings.

```suggestion
    mergedDictionary['ReconFilterType'] !== ['none']
```

Comment (nellh) [reply bids-validator/validators/nifti/nii.js]: Looks like it's string or array of strings, I'll fix that. Thanks!

Comment (nellh) [reply bids-validator/validators/nifti/nii.js]: Okay, now adds the additional required fields for either the string 'none' or an array that has any element 'none'.

**Review by effigies — APPROVED** [review]

**Pull request merged**